common-solidity/coordination-contract/contracts/CrosschainCoordinationV1.sol

```diff
@@ -42,6 +51,24 @@ contract CrosschainCoordinationV1 is CrosschainCoordinationInterface {
         VOTE_CHANGE_PUBLIC_KEY              // 5
     }
 
+    // The public key may be in a state of flux.
+    // So it is important to understand the 'status' of the public key.
+    enum PublicKeyStatus {
+        KEY_CURRENT,                        // 0: This is the active public key for the sidechain which is currently in use
+        KEY_PROPOSED,                       // 1: The public key that has been returned is flagged as a proposed changed key, so it is dependent on the voting before it can become active
+        KEY_PREVIOUS                        // 2: This is a public that has been used previously, but is currently not in use. Note that this key could be the prior key, or an historic key
+    }
+
+    struct PublicKey {
+    //  uint sidechainID;   // Now that I added the struct to sidechains record, we probably don't need this
+        uint versionNumber;
+        uint256 status;
```

Comment:
I think the status should be able to fit into a field smaller than a uint256. If we allow it to be a uint64, then that will allow for a huge variety of status codes, and will combine with the uint64 field above to compress when stored to a single uint256 storage slot.

Reply:
Changed to uint64